sympy/solvers/solveset.py

```diff
@@ -209,6 +209,32 @@ def _invert_complex(f, g_ys, symbol):
                                                log(Abs(g_y))), S.Integers)
                                for g_y in g_ys])
             return _invert_complex(f.args[0], exp_invs, symbol)
+
+    if isinstance(f, (sin, csc)):
+        n = Dummy('n')
+        if isinstance(g_ys, FiniteSet):
+            fi = sympify('a' + f.func.__name__)
```

Comment:
Please don't use strings to do symbolic manipulation. Trig functions have `inverse` method for exactly this.